src/main/data-dir.ts

```diff
@@ -4,10 +4,12 @@ import path from 'path'
 import {promises as fs, constants as fsConstants} from 'fs'
 import {satisfies, coerce} from 'semver'
 import rimraf from 'rimraf'
-import {dialog} from 'electron'
+import {BrowserWindow, dialog} from 'electron'
 import {DATADIR_VERSION, COMPATIBLE_VERSIONS} from '../shared/version'
-import {createAndInitI18nForMain, createTFunctionMain} from './i18n'
-import {DEFAULT_LANGUAGE} from '../shared/i18n'
+import {TFunctionMain} from './i18n'
+import {isDirEmpty} from './util'
+
+const NODE_DIR_NAME = 'Mantis'
```

Comment:
What is it for?

Reply:
The user selects a folder where the `Mantis` folder will be placed.
e.g. if the user selects `~/.mantis-wallet-dev`, the node data will be stored in `~/.mantis-wallet-dev/Mantis`.
(there's a check if the basename is `Mantis`, i.e. if the user selects `~/.mantis-wallet-dev/Mantis`, then the node folder will be the same.

Reply:
@qwhex Maybe we can just remove this subfolder creation and let the user directly pick the folder where mantis data is. Right now, with this extra folder created, all existing users will be forced to either sync again or to manually move their existing mantis folder. Also, another usecase is when user already has mantis client running and wants to connect wallet to it. With this extra folder, user has no other option, just to manually move the data after changing datadir.

Reply:
I think by simply dumping the data in the folder which the user selects more easily leads to problems.

One example is that the user selects his home directory thinking that the data will be put there in a subfolder. By your suggestion, the home folder itself will be used as datadir - it's not empty but "incompatible", then if the user selects the delete option: oops. (Although there's a clear indication of which folder will be deleted / an extra confirmation, nonetheless: this could happen)

I prefer to keep this extra safety measure to not accidentally trash the user's folders.

By changing it from `Mantis` to `mantis`, the default datadir location will be the same between the two versions. (`~/.mantis-wallet/mantis`)

I can add some extra logic to ease the migration from the previous version - i.e. checking parent folder (wallet datadir in default case) for version.txt and copying it to the mantis datadir.

Reply:
Yeah, I would definitely prefer `mantis` :) Actually, we could even make it super-smart: after selecting directory, check whether it's mantis datadir, if yes, use it, if not, create `mantis` subfolder. Not sure how to do that check properly though.

Reply:
But in this way it isn't really compatible with case where user runs mantis through CLI and then runs `Mantis-Wallet`, right? Simply because there's no way to select exactly the same directory

Reply:
> I think by simply dumping the data in the folder which the user selects more easily leads to problems.
Please note that whole feature is aimed at more advanced users anyway. If they make a mess in their directory structure, it's on them.

Reply:
It's not aimed at advanced users. 😅 How is it advanced to select where to put your data if we're talking about hundreds of gigabytes? For Mainnet users, it's an essential feature.

---

If the user creates a datadir with the CLI, it won't create subdirs by `networkName`, right?
If so, whatever we do, the user has to move/rename the folders anyway.

Otherwise, I'd go with detecting if the selected folder is already a datadir, i.e. checking if it contains one of the proper subdirs, i.e `datadir/testnet-internal-nomad`. and not appending `mantis` to the path if so.

Reply:
@qwhex @kapke How about asking user whether he wants to create `mantis` subfolder under selected location using some simple prompt? Would everyone be happy with that? 😄

Reply:
I think it doesn't matter if by creating datadir on the CLI there are no subdirs like `testnet-internal-nomad`. It's simply not compatible with the network switcher.

I'd go for updating the documentation on how to use the datadir created with the CLI.
i.e. renaming your datadir to the e.g. `testnet-internal-nomad` and moving it to a folder named `mantis`.
After these steps, everything is handled by the wallet (adding version file / using the data / placing logs in proper dirs)

BTW Mantis-Wallet already detects if the selected folder is named `mantis`, so it doesn't matter if you select `.mantis-wallet` or `.mantis-wallet/mantis`